evalml/pipelines/utils.py

```diff
@@ -657,6 +669,11 @@ def _make_new_component_name(name, component_name, idx=None, pipeline_name=None)
             new_component_name = _make_new_component_name(
                 component_pipeline_name, name, name_idx, sub_pipeline_name
             )
+            first_x_component = (
```

Comment:
I think there might be a bug in the split pipeline creation.

```python
from evalml.demos import load_fraud
from evalml.automl import AutoMLSearch
from evalml.pipelines.components import DFSTransformer
from evalml.exceptions import AutoMLSearchException
import pytest

X, y = load_fraud(1000)

dfs = DFSTransformer()

# Case 1: Including identity feature in X_train
X_t = dfs.fit_transform(X)
X_t.ww['datetime'] = X['datetime']
automl = AutoMLSearch(X_t, y, "binary", _automl_algorithm="default",
                      max_batches=4, features=dfs.features, verbose=True)

with pytest.raises(Exception, match="All pipelines in the current AutoML batch"):
    automl.search()
    
# Case 2: Not including identity feature in X_train
automl = AutoMLSearch(X, y, "binary", _automl_algorithm="default",
                      max_batches=4, features=dfs.features, verbose=True)
with pytest.raises(Exception,
                   match="not found in DataFrame",
                   ):
    automl.search()
```

These two cases pass for the iterative algorithm.

Reply:
thanks! will take a look, still stuck on running integration tests so I'm sure theres more breaking!